src/multicorn.c

```diff
@@ -581,7 +572,24 @@ multicornGetForeignPlan(PlannerInfo *root,
     /* Extract data needed for aggregations on the Python side */
     if (IS_UPPER_REL(foreignrel))
     {
+        /*
+         * TODO: fdw_scan_tlist is present in the execute phase as well, via
+         * node->ss.ps.plan.fdw_scan_tlist, and instead of root, one can employ
+         * rte = exec_rt_fetch(rtindex, estate) to fetch the column name through
+         * get_attname.
+         *
+         * Migrating the below function into multicornBeginForeignScan would thus
+         * reduce the duplication of plan and execute fields that are now being
+         * serialized.
+         */
         multicorn_extract_upper_rel_info(root, fdw_scan_tlist, planstate);
+
+        /*
+         * Since scan_clauses are empty in case of upper relations for some
+         * reason. We pass the clauses from the base relation obtained in MulticornGetForeignRelSize.
+         */
+        ofpinfo = (MulticornPlanState *) planstate->outerrel->fdw_private;
+        planstate->baserestrictinfo = extract_actual_clauses(ofpinfo->baserestrictinfo, false);
```

Comment:
Just to check my understanding:

  - This code used to split `baserestrictinfo` into `local_conds` and `remote_conds` with `multicorn_classify_conditions` in `multicornGetForeignRelSize`
  - Instead, we add the full `baserestrictinfo` to the plan state and use it to `extractRestrictions` into the `qual_list` (that makes it to Python)

observations/questions:
  - as far as I can see, `remote_conds` isn't actually used anywhere on master, so removing it has no effect
  - `local_conds` is used here (https://github.com/splitgraph/Multicorn/blob/cbe45ed682dde713aa46a08205ee794c1d6f948a/src/deparse.c#L497) and here (https://github.com/splitgraph/Multicorn/blob/cbe45ed682dde713aa46a08205ee794c1d6f948a/src/multicorn.c#L1686-L1692) but it looks like we never actually populate it after this change? Does that affect anything?

Reply:
Your understanding is correct. And we don't populate `local_conds` so this change basically has no effect.

Effectively, I took the liberty of treating all `baserestrictinfo` as `remote_conds` for now, since `local_conds` come into play with joins, which are not pushed down presently.

Reply:
Got it -- is it worth deleting that code that uses `local_conds` altogether (since those two statements I linked now look like no-ops to me) and removing it from the struct?

Reply:
Looks like with the new approach I'll need the second if statement, need to still rethink it a bit.

Reply:
Ok, I've decided to go with a much simpler implementation in the end, and so was able to remove the `local_conds` completely.